### ISSUE TITLE
publish --missing=inherit won't work in general

### ISSUE THREAD

**Issue by bpoldrack** [datalad/datalad#1344]
The option `--missing=inherit` was introduced for `publish`. As of PR #1299, this leads to a call to `create-sibling` with settings inherited from the superdataset if a dataset is missing the sibling in question.
While it's okay to have this and play with it a little bit, this approach won't stand reality ;-)
As I pointed out in #1299, I think this should either be just `add-sibling` or we an even more complex strategy altogether.
The reason is, that first of all `create-sibling` might easily fail due to missing permissions of the publisher,second it's somewhat weird to let datalad's counterpart of `git push` actually create a new repository at the remote and third (may be even most important): We already have _two_ `create-sibling` commands and there may well be more in the future. By now, `publish` can't decide which kind of `create-sibling` is needed.

Bottom line: Two possibilities from my point of view:

- keep it simple and only allow for `add-sibling` from within `publish`
- `create-sibling` must leave an indication of itself (which kind) in the repo's `.datalad/config` and `publish` then would need to read that and call the respective command. Even then, `publish` would need to just try creating the thing, if it fails at least add it, then try using it to publish and if still fails ... well, skip_failed or even more options to set ...

**Comment by yarikoptic** [comment]
> While it's okay to have this and play with it a little bit, this approach won't stand reality ;-)

it is a funny one since I have been using it in reality for a while.  There are problems of cause (e.g. #1491) but nothing which could not be solved.  If it cannot create on remote end -- it rightfully could/should fail.

**Comment by mih** [comment]
Also see #1622

**Comment by mih** [comment]
I am closing this one. It will have the same archival value with the new state.
